TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Service/ContentProxyableEntityService.php

```diff
@@ -0,0 +1,209 @@
+<?php
+namespace TYPO3\TYPO3CR\Domain\Service;
+
+/*
+ * This file is part of the TYPO3.TYPO3CR package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use TYPO3\Flow\Annotations as Flow;
+use TYPO3\Flow\Log\SystemLoggerInterface;
+use TYPO3\Flow\Object\ObjectManagerInterface;
+use TYPO3\Flow\Persistence\PersistenceManagerInterface;
+use TYPO3\Flow\Reflection\ObjectAccess;
+use TYPO3\Flow\Reflection\ReflectionService;
+use TYPO3\Flow\Utility\Arrays;
+use TYPO3\TYPO3CR\Domain\Factory\NodeFactory;
+use TYPO3\TYPO3CR\Domain\Model\ContentProxyProxyableEntityInterface;
+use TYPO3\TYPO3CR\Domain\Model\NodeInterface;
+use TYPO3\TYPO3CR\Domain\Repository\ContentProxyableRepository;
+use TYPO3\TYPO3CR\Domain\Repository\NodeDataRepository;
+use TYPO3\TYPO3CR\Exception;
+
+/**
+ * Manage synchronization between the content repository and doctrine entities
+ *
+ * @Flow\Scope("singleton")
+ * @api
```

Comment:
Might also need an interface at some point or strategies depending on entity. Think if we add CQRS/ES for entities or someone stores them not in doctrine.